src/Bicep.Cli/Commands/BuildCommand.cs

```diff
@@ -76,7 +76,15 @@ public async Task<int> RunAsync(BuildArguments args)
                 return diagnosticLogger.ErrorCount > 0 ? 1 : 0;
             }
 
-            logger.LogError(CliResources.UnrecognizedFileExtensionMessage, inputPath);
+            if(!features.ParamsFilesEnabled && IsBicepparamsFile(inputPath)) 
+            {
+                logger.LogError("Please enable the parameters file feature flag to compile the parameter file");
```

Comment:
Any suggestions on best practice for throwing this error?

Reply:
Should this be raised as an error diagnostic from within Bicep.Core instead? Anything we check in a CLI command also needs to be checked where the language server attempts the same action.

Reply:
No, this is deliberately just a CLI UX change to unblock AzureCLI & AzurePowershell implementations.

The goal is to be able to to have the following logic in AzureCLI, so we're just trying to ensure the error message is intuitive enough to allow AzureCLI/AzurePowershell to delegate to Bicep CLI:
```
if (bicep_version < <jan_release_version_number>)
  log_generic_bicep_params_not_supported_error()
else
  invoke_bicep_cli()
```

This unblocks us to start on the AzureCLI/AzurePowerShell implementation immediately (instead of having to wait for BicepCLI to have full support for BicepParams)

Reply:
@Usman0111 I would recommend following the pattern we've used for the `UnrecognizedFileExtensionMessage` and adding an entry to `src\Bicep.Cli\CliResources.resx` - this allows us to support localization in future if necessary. I can share some pointers offline.

Reply:
That concern seems orthogonal to the question of where this error is raised. I'm guessing we have separate flag checking for the language server, but both could be consolidated into `SemanticModel` by adding a `GetFileTypeDiagnostics` called from `AssembleDiagnostics` with a body along the lines of:

```csharp
private IEnumerable<IDiagnostic> GetFileTypeDiagnostics()
{
    if (SourceFile.FileKind == BicepSourceFileKind.ParamsFile && !Features.ParamsFilesEnabled)
    {
        yield return DiagnosticBuilder.ForDocumentStart().<method to create this error diagnostic>();
    }
}
```

Not required for this PR, but I am curious as to what happens with this PR in place when a .bicepparam file is built using a VS code task rather than the CLI.